metricflow/dataflow/dataflow_plan.py

```diff
@@ -718,6 +718,7 @@ class JoinToTimeSpineNode(Generic[SourceDataSetT], BaseOutput[SourceDataSetT], A
     def __init__(
         self,
         parent_node: BaseOutput[SourceDataSetT],
+        metric_time_dimension_spec: TimeDimensionSpec,
```

Comment:
Do we want this here?

Previously, the granularity was sourced from the parent node for making the time spine dataset, which contained all of the available granularities. Now it's coming in from the query parameters.

Now it's getting threaded through from the callsite, and what we're doing at the callsite is taking one of the potentially many query instances and wiring it through this node.

Reply:
The parent dataset will no longer be narrowed down to only the requested `metric_time_dimension_instances` since the measure aggregation is being moved to after joining to time spine in this PR. Because of that, I thought it would be more efficient to pass through the requested granularity and only aggregate time spine columns based on that request. However, I can see a couple of potential issues with that.
1. There might be multiple `metric_time` dimensions in the query. To handle this case, we could accept a list of `metric_time_dimension_instances` in `_make_time_spine_data_set` and create `DATE_TRUNC` columns for each.
2. Maybe we don't want to prematurely optimize this query, and I should leave that to the optimizer. In that case we could create a `DATE_TRUNC` column for every possible granularity in `_make_time_spine_data_set`, and expect the unrequested ones to get filtered out by `FilterElementsNode`.

I'll definitely need to update the logic to handle case #1, but not sure about #2. Thoughts on that one?

Reply:
Actually, after some noodling on this I understood more what you meant. Ended up going back to inheriting the lowest granularity available from the parent dataset after all. So you can ignore this thread!